Source/common/CEL.cc

```diff
@@ -0,0 +1,236 @@
+/// Copyright 2025 North Pole Security, Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///    http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+
+#include "Source/common/CEL.h"
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "cel/expr/checked.pb.h"
+#include "common/ast_proto.h"
+#include "compiler/compiler_factory.h"
+#include "compiler/standard_library.h"
+#include "eval/public/activation_bind_helper.h"
+#include "eval/public/builtin_func_registrar.h"
+#include "eval/public/cel_expr_builder_factory.h"
+#include "eval/public/cel_function_adapter.h"
+
+namespace cel_runtime = ::google::api::expr::runtime;
+namespace syncv1 = ::santa::sync::v1;
+namespace pbv1 = ::santa::cel::v1;
+
+namespace santa {
+
+// Creates a CEL Activation with the static and dynamic contexts attached
+// appropriately.
+absl::StatusOr<cel_runtime::Activation> CreateActivation(
+    ::pbv1::CELStaticContext *static_context,
+    ::pbv1::CELDynamicContext * (^dynamicBlock)(
+        ::google::protobuf::Arena *arena)) {
+  cel_runtime::Activation activation;
+
+  if (dynamicBlock) {
+    absl::StatusOr<std::unique_ptr<cel_runtime::CelFunction>> getArgsFunction =
+        cel_runtime::FunctionAdapter<::pbv1::CELDynamicContext *>::Create(
+            "getDynamic", false,
+            std::function<::pbv1::CELDynamicContext *(
+                ::google::protobuf::Arena *)>(
+                [dynamicBlock](::google::protobuf::Arena *arena)
+                    -> ::pbv1::CELDynamicContext * {
+                  return dynamicBlock(arena);
+                }));
```

Comment:
nit: This can be simplified, the `std::function` should be able to be removed and the type deduced from the lamba which would remove the duplication. You should also be able to just use block syntax to keep things consistent with the rest of the file.
```
^::pbv1::CELDynamicContext *(::google::protobuf::Arena *arena) {
    return dynamicBlock(arena);
}
```